tst/unit/kokkos_abstraction.cpp

```diff
@@ -33,405 +41,513 @@ using parthenon::ParArray3D;
 using parthenon::ParArray4D;
 using Real = double;
 
-template <class T>
-bool test_wrapper_1d(T loop_pattern, DevExecSpace exec_space) {
-  // https://en.cppreference.com/w/cpp/numeric/random/uniform_real_distribution
-  std::random_device rd;  // Will be used to obtain a seed for the random number engine
-  std::mt19937 gen(rd()); // Standard mersenne_twister_engine seeded with rd()
-  std::uniform_real_distribution<Real> dis(-1.0, 1.0);
-
-  const int N = 32;
-  ParArray1D<Real> arr_dev("device", N);
-  auto arr_host_orig = Kokkos::create_mirror(arr_dev);
-  auto arr_host_mod = Kokkos::create_mirror(arr_dev);
-
-  // initialize random data on the host not using any wrapper
-  for (int i = 0; i < N; i++)
-    arr_host_orig(i) = dis(gen);
+template <std::size_t Ni>
+using Sequence = std::make_index_sequence<Ni>;
+
+template <std::size_t>
+struct ParArrayND_impl {};
+template <>
+struct ParArrayND_impl<0> {
+  template <typename T>
+  using type = parthenon::ParArray0D<T>;
+};
```

Comment:
It might be worth finding a name for this and moving it into the source. You might not be the first one to want the ability to select the dimension via a `size_t`.

Reply:
There is a function buried in there for getting one of these that could be what actually gets exposed.
```cpp
template <std::size_t ND, typename T, typename... Args>
auto ParArrayND(Args &&...args) {
  static_assert(ND <= 8, "ParArrayND supoorted up to ND=8");
  return typename ParArrayND_impl<ND>::template type<T>(std::forward<Args>(args)...);
}
```

Reply:
Ah I see. Yes maybe worth naming that internal type something more useful so we can expose it if we want to. `ParArray` without the `ND` maybe... or something like that.